site/topics/objects/methods.rst

```diff
@@ -17,10 +17,59 @@ Objects II --- More on Methods
 Point3D Class
 =============
 
+* We will start with the ``Point3D`` class since it will be utilized by the ``Sphere`` class
```

Comment:
This was the original plan, but I went with the bottom up approach since, although it's not too bad saying something like 
      - "In the `Sphere` constructor, we will want a way to keep track of a _point_, so we may want to consider a class for this"
      - "Hmm, in the `distance_between_centres` method, if our point class has a method for calculating their distances, that's great!"
      - etc. 

The trouble arises specifically with the `__eq__` method since it would be (a) introducing this whole new idea, and (b) need to be referring to the potential point class too.  All that said, now that I "articulate it" (type it), it really doesn't seem like that big of a deal. 

I suppose the easiest option at this stage would be to _design_ the high-level idea of what we want the `Sphere` class to do with no real code anywhere, just pseudo-pseudo-code. For example:
    - We want a class to keep track of `Sphere`s in space, so we need a radius and a way to represent a point
    - We want to check the distances between the centres of the spheres --- hmmm, neato, if our points can know their distances from one another, done-zo.
    - Distances between edges, well, distances between centres minus radii
    - Equality, are the radii equal and centre points?
    - Ok great, now that we know what we want, let's think about that _point_ 
    - (do `Point3D`)
    - Now do `Sphere` 

If this is the approach taken, then I think the best way to move forward is:
    - Break this topic into two --- (first half is design of `Sphere` and them imp of `Point3D`, second is imp of `Sphere`)
    - Add a section before `Point3D` doing the design of the `Sphere` 
    - Adjust `Point3D` content already made accordingly

Reply:
You're approaching it with a design already in mind. That won't be the case in any novel problem space. Your initial design could be the sphere does everything. Give it `(x,y,z)` properties, write the whole class, realize it's crap, then extract out a reasonable component from it.

Reply:
yooooooo

Reply:
Current Idea:
Objects II --- Methods 
    - Make the `Sphere` class with x, y, z, attributes
 
Objects III --- Interacting Objects
    - Reflect that, although the `Sphere` class works the way it is, maybe we can improve 
        - There is a lot of distance between point functionality 
        - If we wanted other shapes, having a point class will come in handy 
    - Implement the `Point3D` class
    - Re-implement the `Sphere` class

Objects IIII ---- Data Structures